src/Compiler/Hoopl/Unique.hs

```diff
@@ -100,8 +100,8 @@ instance IsMap UniqueMap where
 
   mapMap f (UM m) = UM (M.map f m)
   mapMapWithKey f (UM m) = UM (M.mapWithKey (f . intToUnique) m)
-  mapFold k z (UM m) = M.fold k z m
-  mapFoldWithKey k z (UM m) = M.foldWithKey (k . intToUnique) z m
+  mapFold k z (UM m) = M.foldr k z m
+  mapFoldWithKey k z (UM m) = M.foldrWithKey (k . intToUnique) z m
   mapFilter f (UM m) = UM (M.filter f m)
 
```

Comment:
Erikd made a change to fix M.fold and M.foldWithKey.  The change was merged, however  that change didn't fix S.fold.   Please sync with the master, and resubmit the pull request. Thanks.